text/trace/0173-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,242 @@
+state. The goal is to declare messaging semantic conventions stable before the
+end of 2021.
+
+## Motivation
+
+Many observability scenarios involve messaging systems, event streaming, or
+event-driven architectures. For Distributed Tracing to be useful across the
+entire scenario, having good observability for messaging or eventing operations
+is critical. To achieve this, OpenTelemetry must provide stable conventions and
+guidelines for instrumenting those operations.
+
+Bringing the existing experimental semantic conventions for messaging to a
+stable state is a crucial step for users and instrumentation authors, as it
+allows them to rely on [stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability),
+and thus to ship and use stable instrumentation.
+
+## Roadmap
+
+1. This OTEP, consisting of scenarios and a proposed roadmap, is approved and
+   merged.
+2. [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability)
+   for semantic conventions are approved and merged. This is not strictly related
+   to semantic conventions for messaging but is a prerequisite for stabilizing any
+   semantic conventions.
+3. OTEPs proposing guidance for general instrumentation problems that also
+   pertain to messaging are approved and merged. Those general instrumentation
+   problems include retries and instrumentation layers.
+4. An OTEP proposing a set of attributes and conventions covering the scenarios
+   in this document is approved and merged.
+5. Proposed specification changes are verified by prototypes for the scenarios
+   and examples below.
+6. The [specification for messaging semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md)
+   are updated according to the OTEP mentioned above and are declared
+   [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable).
+
+## Terminology
+
+To leverage existing standards, the terminology used in this document is based
+on the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md).
+CloudEvents is hosted by the CNCF and provides a specification for describing
+event data in common formats to provide interoperability across services,
+platforms and systems.
+
+### Message
+
+A "message" is a transport envelope for the transfer of information. The
+information is a combination of a payload and metadata.  Messages may be
+transferred directly between parties or via one or more intermediaries.  A
+message may carry annotations that are directed at intermediaries on the
+message path.  Messages are uniquely identifiable.
```

Comment:
Does this mean each message has a unique ID or is this meant to be more conceptual (i.e., that each message is an individual with individual properties).

Reply:
More the former ("each message has a unique ID"). I'm not quite sure what the latter ("each message is an individual with individual properties") would mean, sounds to me like "each message is unique", which doesn't make much sense.

The idea behind this is, that when one looks at messages in different places (producers, consumers, observability backends), one can tell whether the messages are the same or not.

It is put in a more opaque way, as "a unique ID" suggests a single attribute, whereas a message can also be uniquely identifiable by a combination of attributes.